data-prepper-plugins/csv-processor/src/main/java/com/amazon/dataprepper/plugins/processor/csv/CSVProcessor.java

```diff
@@ -45,4 +111,44 @@ public boolean isReadyForShutdown() {
     public void shutdown() {
 
     }
+
+    private CsvMapper createCsvMapper() {
+        CsvMapper mapper = new CsvMapper();
+        mapper.enable(CsvParser.Feature.WRAP_AS_ARRAY); // allows mapper to read with empty schema
+        return mapper;
+    }
+
+    private CsvSchema createCsvSchema() {
+        char delimiter_as_char = this.delimiter.charAt(0); // safe due to config input validations
+        char quote_char_as_char = this.quote_char.charAt(0); // safe due to config input validations
+        CsvSchema schema = CsvSchema.emptySchema().withColumnSeparator(delimiter_as_char).withQuoteChar(quote_char_as_char);
+        return schema;
+    }
+
+    private void putDataInEvent(Event event, List<String> header, List<String> data) {
+        // open question: We decided that if there is a user-specified header and a header source event key then we use the header source
```

Comment:
We shouldn't commit a comment like this. This would be a good thing to move to the Github issue

Reply:
Good catch! I removed the code comment and clarified in the GitHub issue (#1081).